spec/community.mediawiki

```diff
@@ -55,3 +55,34 @@ of any unfilled orders.
 
 Less drastic punitive measures such as a cool-down period may be considered
 for minor, first-time or infrequent conduct violations.
+
+===Penalization Notification===
+
+In the event that a rule is broken, the client will be sent a notification that
+includes penalization details. If the client is offline, this information will
+be sent in the [[comm.mediawiki/#Session_Authentication|connect response]].
+
+'''Notification route:''' <code>penalty</code>, '''originator: ''' DEX
+
+<code>payload</code>
+{|
+! field   !! type   !! description
+|-
+| penalty || object || a Penalty object (definition below)
+|}
+
+
+====Penalty object====
+{|
+! field      !! type   !! description
+|-
+| brokenrule || int    || the rule that was broken
+|-
+| timestamp  || int    || the UNIX timestamp in milliseconds at which the penalization occured
+|-
+| duration   || int    || the duration in milliseconds of the penalization
+|-
+| details    || string || a message describing the penalization
+|-
+| sig        || string || hex-encoded server's signature of the serialized penalization data
+|}
```

Comment:
You'll need to describe the serialization.

Reply:
As in how many bytes per field and the order?

Reply:
I found a few different patterns, but will just adding the size of the fields be sufficient in this case? Should there be a separate `Penalty serialization`?

Reply:
I'd personally prefer a serialization for the `Penalty` object. For the `penalty` payload you wouldn't need a whole serialization description table, because you could just say **sig || string || hex-encoded signature of the serialized `Penalty` object.**

> I found a few different patterns, but will just adding the size of the fields be sufficient in this case?

In this case, since all of the fields of the `Penalty` object are included in the serialization, you could try to combine them into one. I think the only place that's done so far is for the order prefix description, and I have mixed feelings on how well it works. The descriptions needed for serialization is often quite different than a description suited for the JSON field, so you'll need to combine the information carefully to avoid confusion.